### DISCUSSION TITLE
How to combine livewire and an input mask library such as inputmask

### DISCUSSION THREAD

**Discussion by MarijnRitzen** [livewire/livewire#4616] [1. Help]
So I'm trying to use an inputmask on the frontend (https://github.com/RobinHerbots/Inputmask). 
However, the problem is that specifiying this input mask on the inputs does not go well with wire:model. 
I think this has to do with the fact that livewire replaces the input fields on every roundtrip to the server, and thus I have to call the inputmask code on the new input field every time. Here is the code to replicate:

Test.php (livewire component server side)
```
namespace App\Http\Livewire;

use Livewire\Component;

class Test extends Component
{
    public $vat = 0;
    public $withvat = 0;
    public $withoutvat = 0;

    public function updatedVat($value) {
        if ($value === "" || !is_numeric($value)) {
            $value = 0;
        }

        $this->withoutvat = $this->withvat / (1 + $value / 100);
    }

    public function updatedWithvat($value) {
        if ($value === "" || !is_numeric($value)) {
            $value = 0;
        }

        $this->withoutvat = $value / (1 + $this->vat / 100);
    }

    public function updatedWithoutvat($value) {
        if ($value === "" || !is_numeric($value)) {
            $value = 0;
        }

        $this->withvat = $value * (1 + $this->vat / 100);
    }

    public function render()
    {
        return view('livewire.test')->extends('layouts.app');
    }
}
```

livewire component frontend: 
```
<div>
    <label for="vat">Vat:</label>
    <input type="number" id="vat" wire:model="vat">

    <label for="withVat">Price with vat:</label>
    <input type="text" id="withVat" class="money" wire:model="withvat">

    <label for="withoutVat">Price without vat:</label>
    <input type="text" id="withoutVat" class="money" wire:ignore wire:model="withoutvat">

    <script>
        $(document).ready(function () {
            $(".money").inputmask("9{+},99", {
                positionCaretOnClick: "none",
                autoUnmask: true,
                radixPoint: ",",
                numericInput: true,
                placeholder: "0",
            });
        });
    </script>
</div>
```

Removing the `$(".money").inputmask function makes it work with cents. But in order to make it work with actual euros I want to use the inputmask. However changing something with the inputmask does not even send a request to the server. Maybe because of overwritten or conflicting input listeners?

**Comment by joshhanley** [comment]
@MarijnRitzen yeah if you use javascript libraries that modify the DOM, when Livewire does an update, it will wipe out any of those modifications (as it doesn't know about them).

To get around this, you need to make use of `wire:ignore`, see here in the docs https://laravel-livewire.com/docs/2.x/alpine-js#ignoring-dom-changes

Due to that not being controlled by Livewire, you may need to make use of Alpine to sit between your library and Livewire and use it to pass data on. Have a look at using `@entangle` https://laravel-livewire.com/docs/2.x/alpine-js#sharing-state

Hope this helps!

**Comment by hammadshah01** [comment]
Well Guys in Laravel Livewire 3 Input masking is too easy 

**Step1: (INSTALL THIS PACKAGE)** 

`composer require hallindavid/manny`

**Step2: (Create a function under your mount function or any where on top)**

```
public function updated($field)
 {
 if ($field == 'cnic') { $this->cnic = Manny::mask($this->cnic, "11111-1111111-1"); 
}
```
 
**For Livewire Blade it would be like this**
```
  <input type="text"  id="cnic"
                class="form-control rounded-0" placeholder="00000-0000000-0" wire:model.live="cnic">
```
use _**wire:model.live**_ or _**_wire:model.lazy_**_ choice is yours
